tests/integration/models/reddit/test_submission.py

```diff
@@ -380,6 +380,14 @@ def test_send_removal_message(self, _):
             assert res[1] is None
             assert res[2] is None
 
+    @mock.patch("time.sleep", return_value=None)
+    def test_set_original_content(self, _):
+        self.reddit.read_only = False
+        with self.recorder.use_cassette(
+            "TestSubmissionModeration.test_set_original_content"
+        ):
+            Submission(self.reddit, "dueqm6").mod.set_original_content()
```

Comment:
It would be nice if this test checks that the submission is actually marked as original content. To do this, I would first say `submission = Submission(self.reddit, "dueqm6")` and then check that it isn't original content, and then mark it, and then check that it is. Before running this test, make sure that the submission isn't marked as OC.

This would look something like

```python
submission = Submission(self.reddit, "dueqm6")
assert not submission.is_original_content
submission.mod.set_original_content()
assert submission.is_original_content
```

Make sure to delete the existing cassette before running the test, so that the cassette can be re-recorded for the new test.

Reply:
So I implemented those changes (also chose a different post for the `unset` method) but now it's throwing an error even though I'm certain the post `dueqm6`'s attribute `is_original_content` evaluates to `False` as the post is *not* set as OC.

```
________________________________________________ TestSubmissionModeration.test_set_original_content _________________________________________________

self = <tests.integration.models.reddit.test_submission.TestSubmissionModeration object at 0x103d2ee50>, _ = <MagicMock name='sleep' id='4358430224'>

    @mock.patch("time.sleep", return_value=None)
    def test_set_original_content(self, _):
        self.reddit.read_only = False
        with self.recorder.use_cassette(
            "TestSubmissionModeration.test_set_original_content"
        ):
            submission = Submission(self.reddit, "dueqm6")
            assert not submission.is_original_content
            submission.mod.set_original_content()
>           assert submission.is_original_content
E           AssertionError: assert False
E            +  where False = Submission(id='dueqm6').is_original_content

tests/integration/models/reddit/test_submission.py:392: AssertionError
```

Reply:
If it works on your machine but fails only on Travis, then you just need to upload the cassettes as before, because they have changed. If it fails on your machine as well, note that you need to delete the existing cassettes (just the two that you created) so that they will be re-recorded. Then, commit and push the working cassettes.

Reply:
It's failing on my machine as well. I've deleted the cassettes that were previously generated.

Reply:
I'll also note that the equivalent `nsfw()` and `spoiler()` tests don't have `assert` checks. 🤔

Reply:
I think I know how to solve it. I need to refresh the submission after the method call:

```
submission = Submission(self.reddit, "dueqm6")
assert not submission.is_original_content
submission.mod.set_original_content()
submission = Submission(self.reddit, "dueqm6")
assert submission.is_original_content
```

Reply:
> I'll also note that the equivalent `nsfw()` and `spoiler()` tests don't have `assert` checks. 🤔

Haha, thanks for pointing that out. I think that that isn't great testing form, but AFAIK we haven't actually had any issues with those methods (fingers crossed). I'm always in favor of high standards, so this feature should have good tests even if previous ones don't. 😃

Reply:
> I think I know how to solve it. I need to refresh the submission after the method call:


You're right. Sorry that I didn't catch that.

Reply:
Maybe for my next pull request I'll add the same kind of test to `nsfw()` and `spoiler()`.

Reply:
It's quite old, but see [this comment](https://github.com/praw-dev/praw/pull/772#discussion_r109809764) by @bboe - that the older tests don't re-fetch the objects is the result of a conscious decision.

Reply:
@leviroth Thanks for the reference.